packages/buidler-core/src/internal/buidler-evm/provider/modules/evm.ts

```diff
@@ -32,6 +41,29 @@ export class EvmModule {
     throw new MethodNotFoundError(`Method ${method} not found`);
   }
 
+  // evm_setNextBlockTimestamp
+
+  private _setNextBlockTimestampParams(params: any[]): [number] {
+    return validateParams(params, t.number);
+  }
+
+  private async _setNextBlockTimestampAction(
+    timestamp: number
+  ): Promise<string> {
+    const latestBlock = await this._node.getLatestBlock();
+    const increment = new BN(timestamp).sub(
+      new BN(latestBlock.header.timestamp)
+    );
+    if (increment.lte(new BN(0))) {
+      throw new InvalidInputError(
+        `Timestamp ${timestamp} is lower than previous block's timestamp` +
+          `${new BN(latestBlock.header.timestamp).toNumber()}`
+      );
+    }
+    await this._node.setNextBlockTimestamp(new BN(timestamp));
+    return timestamp.toString();
```

Comment:
Can we return a `numberToRpcQuantity(timestamp)` instead? Most RPC methods encode numbers that way, so I'd rather keep this consistent with them.

Reply:
This is to be consistent with `_increaseTimeAction` as well.